src/Diagnostics.RuntimeHost/Controllers/ProcessHealthController.cs

```diff
@@ -13,18 +12,22 @@ public class ProcessHealthController : Controller
     {
         private ISourceWatcherService _sourceWatcherService;
         private ICompilerHostClient _compilerHostClient;
+        private IHealthCheckService _healthCheckService;
 
-        public ProcessHealthController(ISourceWatcherService sourceWatcherService, ICompilerHostClient compilerHostClient)
+        public ProcessHealthController(ISourceWatcherService sourceWatcherService, ICompilerHostClient compilerHostClient, IHealthCheckService healthCheckService)
         {
             // These dependencies are injected for the services to start.
             _sourceWatcherService = sourceWatcherService;
             _compilerHostClient = compilerHostClient;
+            _healthCheckService = healthCheckService;
         }
 
         [HttpGet(UriElements.HealthPing)]
         public async Task<IActionResult> HealthPing()
         {
             await _sourceWatcherService.Watcher.WaitForFirstCompletion();
+            await _healthCheckService.RunHealthCheck(true);
```

Comment:
Can we make these concurrent calls? `var healthTasks = arrary[sourcewatchertest, outboundtest]...Task.WhenAll(healthTasks)`